packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -25,24 +30,100 @@ const notFinished: AWSCDKAsyncCustomResource.IsCompleteResponse = {
   IsComplete: false,
 };
 
+// #region Entry points
+export const onEvent = cfnResponse.safeHandler(onEventHandler);
+export const isComplete = cfnResponse.safeHandler(isCompleteHandler);
 /**
- * Util function to log especially for nested objects and arrays
- * @param msg
- * @param other arrays of arguments to be logged
+ * Handler for requests sent from CloudFormation for custom resource.
+ * This is the entry point of the custom resource.
+ *
+ * Note: This is adapted from the AWS CDK's provider framework.
+ * https://github.com/aws/aws-cdk/blob/11621e7/packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts
+ * @param cfnRequest Request received from CloudFormation
  */
-const log = (msg: string, ...other: any[]) => {
-  console.log(
-    msg,
-    other.map((o) => (typeof o === 'object' ? JSON.stringify(o, undefined, 2) : o)),
-  );
-};
+// eslint-disable-next-line func-style
+async function onEventHandler(cfnRequest: AWSLambda.CloudFormationCustomResourceEvent): Promise<void> {
+  const sanitizedRequest = { ...cfnRequest, ResponseURL: '...' } as const;
+  log('onEventHandler', sanitizedRequest);
+
+  cfnRequest.ResourceProperties = cfnRequest.ResourceProperties || {};
+
+  const onEventResult = await processOnEvent(sanitizedRequest);
+  log('onEvent returned:', onEventResult);
```

Comment:
Do we know this result is safe to log?

Reply:
It's [logged by the existing custom resource provider framework](https://github.com/aws/aws-cdk/blob/11621e7/packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts) and [by us in our Lambda function today](https://github.com/aws-amplify/amplify-category-api/blob/8777cd1d9609ef4d85c5ea3c95b249cc13ade6e4/packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts#L64).

The result contains the physical resource ID, DynamoDB table ARN / stream ARN / table name. e.g.
```ts
result = {
  PhysicalResourceId: event.PhysicalResourceId,
  Data: {
    TableArn: describeTableResult.Table.TableArn,
    TableStreamArn: describeTableResult.Table.LatestStreamArn,
    TableName: describeTableResult.Table.TableName,
  },
};
```

To answer your question, yes, I believe it is safe to log.